build/solr/scripts/solr-healthcheck.pl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use JSON;
+use LWP::UserAgent;
+
+my @collections = qw(
+    area
+    artist
+    cdstub
+    editor
+    event
+    instrument
+    label
+    place
+    recording
+    release
+    release-group
+    series
+    tag
+    url
+    work
+);
```

Comment:
Is it needed to check each collection separately?

Maybe a collection-agnostic query can be made instead? [`OVERSEERSTATUS`](https://solr.apache.org/guide/solr/latest/deployment-guide/cluster-node-management.html#overseerstatus) for example?

Otherwise, I would suggest to fetch the list of collections either from API, or from the directory `/usr/lib/mbsssss`.

Reply:
A single API call would be great, though I don't see any fields related to the health or status of the collections in the `OVERSEERSTATUS` response locally, so I'm not sure that one would work.

Getting the list from /usr/lib/mbsssss sounds like a good idea. I avoided using the API in case a collection failed to be created and wasn't returned from the API. In Matrix you mentioned that:

> Solr isn’t accepting connections from outside the container when creating collections.

So I figured we should be sure all of the known collections are active first.

Reply:
From the other discussion:

> To detect whether Solr is running locally or listening to the outside, I didn’t find anything useful in the APIs. This seems to work: `ps auxww | grep Djetty.host=localhost`